lucene/core/src/java/org/apache/lucene/util/IntArrayDocIdSet.java

```diff
@@ -72,7 +72,12 @@ public int nextDoc() throws IOException {
 
     @Override
     public int advance(int target) throws IOException {
-      i = Arrays.binarySearch(docs, i + 1, length, target);
+      int bound = 1;
+      int offset = Math.max(0, i);
+      while(offset + bound < length && docs[offset + bound] < target) {
```

Comment:
I think it's unlikely to happen in practice since we use it for small arrays, but should we protect against integer overflow? Or at least add a comment that explains why this can't overflow?